### PR TITLE
Add packaging command to force a full rebuild

### PR DESCRIPTION
**Pull request by MadCatX**
Lately I have been running into serious difficulties while attempting to build Mol*. Apparently, `tsc` does not build all the modules which leaves me with a barrage of exceptionally useless error messages such as:

```
[webpack-cli] Error: ENOENT: no such file or directory, open '/home/madcat/Devel/BTU_FGU/molstar/lib/mol-plugin/version.js'
    at Object.openSync (node:fs:585:3)
    at Object.writeFileSync (node:fs:2153:35)
    at VersionFilePlugin.apply (/home/madcat/Devel/BTU_FGU/molstar/webpack.config.common.js:10:12)
    at createCompiler (/home/madcat/Devel/BTU_FGU/molstar/node_modules/webpack/lib/webpack.js:73:12)
    at /home/madcat/Devel/BTU_FGU/molstar/node_modules/webpack/lib/webpack.js:44:48
    at Array.map (<anonymous>)
    at createMultiCompiler (/home/madcat/Devel/BTU_FGU/molstar/node_modules/webpack/lib/webpack.js:44:33)
    at create (/home/madcat/Devel/BTU_FGU/molstar/node_modules/webpack/lib/webpack.js:125:16)
    at webpack (/home/madcat/Devel/BTU_FGU/molstar/node_modules/webpack/lib/webpack.js:142:47)
    at WebpackCLI.f [as webpack] (/home/madcat/Devel/BTU_FGU/molstar/node_modules/webpack/lib/index.js:55:16) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: '/home/madcat/Devel/BTU_FGU/molstar/lib/mol-plugin/version.js'
}
```
or

```
Module not found: Error: Can't resolve '/home/madcat/Devel/BTU_FGU/molstar/lib/apps/viewer/index.js' in '/home/madcat/Devel/BTU_FGU/molstar'
resolve '/home/madcat/Devel/BTU_FGU/molstar/lib/apps/viewer/index.js' in '/home/madcat/Devel/BTU_FGU/molstar'
  using description file: /home/madcat/Devel/BTU_FGU/molstar/package.json (relative path: .)
    Field 'browser' doesn't contain a valid alias configuration
    root path /home/madcat/Devel/BTU_FGU/molstar
      using description file: /home/madcat/Devel/BTU_FGU/molstar/package.json (relative path: ./home/madcat/Devel/BTU_FGU/molstar/lib/apps/viewer/index.js)
        no extension
          Field 'browser' doesn't contain a valid alias configuration
          /home/madcat/Devel/BTU_FGU/molstar/home/madcat/Devel/BTU_FGU/molstar/lib/apps/viewer/index.js doesn't exist
        .js
          Field 'browser' doesn't contain a valid alias configuration
          /home/madcat/Devel/BTU_FGU/molstar/home/madcat/Devel/BTU_FGU/molstar/lib/apps/viewer/index.js.js doesn't exist
        .json
          Field 'browser' doesn't contain a valid alias configuration
          /home/madcat/Devel/BTU_FGU/molstar/home/madcat/Devel/BTU_FGU/molstar/lib/apps/viewer/index.js.json doesn't exist
        .wasm
          Field 'browser' doesn't contain a valid alias configuration
          /home/madcat/Devel/BTU_FGU/molstar/home/madcat/Devel/BTU_FGU/molstar/lib/apps/viewer/index.js.wasm doesn't exist
        as directory
          /home/madcat/Devel/BTU_FGU/molstar/home/madcat/Devel/BTU_FGU/molstar/lib/apps/viewer/index.js doesn't exist
    using description file: /home/madcat/Devel/BTU_FGU/molstar/package.json (relative path: ./lib/apps/viewer/index.js)
      no extension
        Field 'browser' doesn't contain a valid alias configuration
        /home/madcat/Devel/BTU_FGU/molstar/lib/apps/viewer/index.js doesn't exist
      .js
        Field 'browser' doesn't contain a valid alias configuration
        /home/madcat/Devel/BTU_FGU/molstar/lib/apps/viewer/index.js.js doesn't exist
      .json
        Field 'browser' doesn't contain a valid alias configuration
        /home/madcat/Devel/BTU_FGU/molstar/lib/apps/viewer/index.js.json doesn't exist
      .wasm
        Field 'browser' doesn't contain a valid alias configuration
        /home/madcat/Devel/BTU_FGU/molstar/lib/apps/viewer/index.js.wasm doesn't exist
      as directory
        /home/madcat/Devel/BTU_FGU/molstar/lib/apps/viewer/index.js doesn't exist

webpack 5.64.4 compiled with 1 error in 511 ms

assets by status 0 bytes [cached] 1 asset
```

if I allow the build to advance a bit further by disabling the VersionFilePlugin.

When this happens, the `lib` directory contains just a handful of directories

    apps  commonjs  examples  mol-plugin-ui  tests  tsconfig.commonjs.tsbuildinfo

which explains the nature of the problem.

I have had this happen on multiple occasions but I always chalked it up to some kind of one-off flake. Unfortunately, right now this issue renders Mol* entirely unbuildable on one of my machines. Omitting the `--incremental` parameter for `tsc` seems to fix the problem. Once done, `npm run build` starts to work again.

I propose a `npm run rebuild` command to deal with this.

(Should I include this in the CHANGELOG if you consider this for upstreaming?)

### CODE REVIEW

**Comment by dsehnal** [comment]
I've run to build issues from time to time too, so thanks for the investigation and suggestions for improvements.

These seem to happen mostly after updating TypeScript (which happened recently). Removing ``.tsbuildinfo`` files (which are used by the ``--incremental`` flag) in the root dir and the lib dir always helped (which would be equivalent to dropping the ``--incremental`` flag).

How about as an alternative, we add ``clean`` task which deletes the ``lib`` folder and ``tsconfig.tsbuildinfo`` in the root folder, and modify the ``build`` task to do ``"build": "npm run clean && npm run build-tsc ...."``.

The benefit of this approach is also that each time we make a new release, it ensures that the ``lib`` folder has the correct contents without removed files. The downside is slightly longer build times, but I think that's not that big of an issue.

What do you think?

**Comment by MadCatX** [comment]
Actually deleting the stale files is obviously the better thing to do. I added another patch with a cleanup script. `npm run rebuild` now calls the cleanup script before jumping to `npm run build`. In addition to `npm run build` there are now `npm run clean` and `npm run rebuild`. Forcing a full rebuild every time seems a bit heavy-handed to me.

File: package.json [hunk 1]
```diff
@@ -16,6 +16,8 @@
     "test": "npm run lint && jest",
     "jest": "jest",
     "build": "npm run build-tsc && npm run build-extra && npm run build-webpack",
+    "clean": "node ./scripts/clean.js",
+    "rebuild": "npm run clean && npm run build-tsc && npm run build-extra && npm run build-webpack",
```
Comment (dsehnal) [inline package.json:20]: Please change this to ``npm run clean && npm run build``.

File: scripts/clean.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright (c) 2019 mol* contributors, licensed under MIT, See LICENSE file for more info.
```
Comment (dsehnal) [inline scripts/clean.js:2]: 2021.

**Review by dsehnal — COMMENTED** [review]
Thanks for the changes, this is great.

Could you also please change the ``version`` task to run ``rebuild`` instead? Making a new package will take a little longer, but will make sure the lib reflects the source code even if some files have been deleted or moved.

File: package.json (hunk 1 above)
Comment (MadCatX) [reply package.json]: Done

File: scripts/clean.js (hunk 2 above)
Comment (MadCatX) [reply scripts/clean.js]: Done

**Review by dsehnal — APPROVED** [review]
All good, thanks.

**Pull request merged**